cli/azd/pkg/tools/azcli/user_agent_policy.go

```diff
@@ -0,0 +1,27 @@
+package azcli
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+type userAgentPolicy struct {
+	userAgent string
+}
+
+// Policy to ensure the AZD custom user agent is set on all HTTP requests.
+func NewUserAgentPolicy(userAgent string) policy.Policy {
+	return &userAgentPolicy{
+		userAgent: userAgent,
+	}
+}
+
+// Sets the custom user-agent string on the underlying request
+func (p *userAgentPolicy) Do(req *policy.Request) (*http.Response, error) {
+	if strings.TrimSpace(p.userAgent) != "" {
+		req.Raw().Header.Set("User-Agent", p.userAgent)
```

Comment:
I'm curious 🧐... Are we overriding the sdk client telemetry with ours? 

If yes, is there a way to report telemetry for both?

Reply:
Updated to now append our User-Agent when one already exists.

Reply:
@wbreza does this change also covers #720 `Integrate telemetry useragent tracing into golang modules` or is there more to it for @weikanglim?